projects/js-toolkit/packages/portal-base/src/build/bundler2.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import * as babel from '@babel/core';
+import babelPresetEnv from '@babel/preset-env';
+import babelPresetReact from '@babel/preset-react';
+import {Project, format} from '@liferay/js-toolkit-core';
+import fs from 'fs';
+import path from 'path';
+
+import abort from '../util/abort';
+import findFiles from '../util/findFiles';
+import runSass from '../util/runSass';
+import spawn from '../util/spawn';
+
+const {info, print} = format;
+
+export default async function bundler2(project: Project): Promise<void> {
+	fs.mkdirSync(project.build.dir.asNative, {recursive: true});
+
+	copyAssets(project);
+	runSass(project);
+	runCompiler(project);
+	runBundler();
+}
+
+function copyAssets(project: Project): void {
+	const assetFiles = findFiles(project.srcDir, (dirent) => {
+		const lowerCaseName = dirent.name.toLowerCase();
+
+		return (
+			!lowerCaseName.endsWith('.js') && !lowerCaseName.endsWith('.scss')
```

Comment:
Do we need to also copy .css files and possibly images?

Reply:
Yes, that's what we do: copy everything but `.js` and `.scss` files (those are processed with babel and sass so those tools take care of generating them in the build dir).

Notice that the conditions are negated: `lowerCaseName.end != '.js' and lowerCaseName.end != '.scss'`

Reply:
🤦  my apologies. I can't read good

Reply:
NP, negations in logical conditions are seldom a good thing... :roll_eyes: